Sources/SourceKitLSP/SourceKitLSPServer.swift

```diff
@@ -725,15 +727,16 @@ extension SourceKitLSPServer: QueueBasedMessageHandler {
 
     logger.log("Received request \(id): \(params.forLogging)")
 
-    if let textDocumentRequest = params as? any TextDocumentRequest {
+    if self.capabilityRegistry?.clientHasExperimentalCapability(DidChangeActiveDocumentNotification.method) ?? false,
```

Comment:
Shouldn't this be the reverse (ie. only infer the active document if the client doesn't support this notification)?

Reply:
Oh, whoops. That’s also why the tests are failing 😉